tools/toolchains/iar.py

```diff
@@ -74,12 +76,15 @@ def __init__(self, target, notify=None, macros=None, build_profile=None,
         elif target.core == "Cortex-M7F":
             asm_flags_cmd += ["--fpu", "VFPv5_sp"]
             c_flags_cmd.append("--fpu=VFPv5_sp")
-        elif target.core == "Cortex-M23" or target.core == "Cortex-M33" or target.core == "Cortex-M33F":
-            self.flags["asm"] += ["--cmse"]
-            self.flags["common"] += ["--cmse"]
+        elif target.core.startswith("Cortex-M33F"):
```

Comment:
Is this whole section being more complicated than necessary? Current IAR manual says it understands `Cortex-M33.fp`, `Cortex-M7.fp.sp` and `Cortex-M7.fp.dp` as cpu options, so you may not need to be passing separate `--fpu` options at all.

Reply:
Agree, but will leave the ‘fpu’ setting cleanup for another PR. Will have to also consider if fpu is default enabled or not in the core selection. As for Cortex-M33 selection as core fpu and dsp is enabled as default and we need to set .no_dsp instead

Reply:
Ah, I see you going for "no_fp" here: https://github.com/ARMmbed/mbed-os/pull/9433/files.

I don't think that's right - will comment there.